erts/etc/win32/erl.c

```diff
@@ -235,33 +235,51 @@ static wchar_t *find_erlexec_dir(wchar_t *erlpath)
     wchar_t *dir =_wcsdup(erlpath);
     wchar_t *p;
     wchar_t *p2;
+
+    wchar_t *long_dir;
+    DWORD length;
     
     /* Chop of base name*/
     for (p = dir+wcslen(dir)-1 ;p >= dir && *p != L'\\'; --p)
         ;
     *p =L'\0';
     p--;
 
+    length = GetLongPathNameW(dir, NULL, 0);
+    if(length == 0) {
+        free(dir);
+        error("Cannot find erlexec.dll");
+    }
+    long_dir = (wchar_t *) malloc(length * sizeof(wchar_t));
```

Comment:
`length` does not include the null terminator.

```suggestion
    long_dir = (wchar_t *) malloc((length + 1) * sizeof(wchar_t));
```